src/PerfView/Authentication.cs

```diff
@@ -0,0 +1,222 @@
+        /// <param name="enable">True to enable the provider or false to disable it.</param>
+        /// <param name="propertyName">Name of the property that changed for notifying the UI.</param>
+        private void Enable(AuthProviderFlags provider, bool enable, [CallerMemberName] string propertyName = null)
+        {
+            if (enable)
+            {
+                UpdateState(_state | provider, propertyName);
+            }
+            else
+            {
+                UpdateState(_state & ~provider, propertyName);
+            }
+        }
+
+        /// <summary>
+        /// Update the state. If it has changed, notify the UI and write the
+        /// new state back to user configuration.
+        /// </summary>
+        /// <param name="newState">The new state.</param>
+        /// <param name="propertyName">Name of the property that changed for notifying the UI.</param>
+        private void UpdateState(AuthProviderFlags newState, string propertyName)
+        {
+            if (newState != _state)
+            {
+                _state = newState;
+                userConfigData[UserConfigDataKey] = ((double)newState).ToString();
+                PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
+            }
+        }
+    }
+
+    /// <summary>
+    /// Extension methods for <see cref="SymbolReaderHttpHandler"/>.
+    /// </summary>
+    internal static class SymbolReaderHttpHandlerExtensions
+    {
+        /// <summary>
+        /// Configure a <see cref="SymbolReaderHttpHandler"/> from authentication options.
+        /// </summary>
+        /// <param name="handler">The handler.</param>
+        /// <param name="authenticationViewModel">The authentication view model.</param>
+        /// <param name="log">A logger.</param>
+        /// <param name="mainWindow">The main window to use as the parent of any modal dialogs.</param>
+        /// <exception cref="ArgumentNullException">One of the parameters is null.</exception>
+        public static void Configure(this SymbolReaderHttpHandler handler, AuthenticationViewModel authenticationViewModel, TextWriter log, Window mainWindow)
+        {
+            if (handler is null)
+            {
+                throw new ArgumentNullException(nameof(handler));
+            }
+
+            if (authenticationViewModel is null)
+            {
+                throw new ArgumentNullException(nameof(authenticationViewModel));
+            }
+
+            if (mainWindow is null)
+            {
+                throw new ArgumentNullException(nameof(mainWindow));
+            }
+
+            handler.ClearHandlers();
+
+            // The order isn't critical, but we chose to put GCM last
+            // because the user might want to use GCM for GitHub and
+            // Developer identity for Azure DevOps. If GCM were first,
+            // it would always pre-empt the developer identity handler.
```

Comment:
Per our discussion, is this the right ordering given that the delegate chain order is in reverse?

Reply:
I fixed it in SymbolReaderHttpHandler by composing the chain in reverse order.